Create 2020 Focus Blog Post (#362)
ipfs/blog: content/post/082-our-focus-for-2020.md

---
date: 2020-02-10
url: 2020-02-10-our-focus-for-2020
title: Our Focus for 2020
author: Molly Mackinlay
---

_"The mission of IPFS is to create a resilient, upgradeable, open network to preserve and grow humanity’s knowledge."_

It’s a new year, and with it time to update the [IPFS Project Roadmap](https://github.com/ipfs/roadmap) with our focus for 2020. As part of that, we also want to reflect on our successes, challenges, and learnings from 2019 to help us stay on track to achieve our mission, and maximize the value and utility we create in the world. 

> Looking for the TL;DR? Hop right over to the [2020 section of our IPFS Roadmap](https://github.com/ipfs/roadmap#2020-priority) to see our focus and epics for the next year.

## Looking Back at 2019
2019 was a super exciting year for the IPFS Project:
- The IPFS Public Network grew by **30x in 2019**! 🚀
- **💯s of thousands of nodes** are participating in the IPFS Network daily

> - **Millions** of users access the IPFS HTTP Gateway every week 🎉

Author: ```suggestion
- **Millions of users** access the IPFS HTTP Gateway every week 🎉
```

- There are **hundreds of dapps, tools, and projects** in the wider IPFS Ecosystem - including newcomers like [Anytype](https://anytype.io/), [Microsoft ION](https://techcommunity.microsoft.com/t5/azure-active-directory-identity/toward-scalable-decentralized-identifier-systems/ba-p/560168), [Haven](https://gethaven.app/) by OB1, [Brave](https://brave.com/), [3box](https://3box.io/), [EthDNS](https://medium.com/the-ethereum-name-service/ethdns-9d56298fa38a) and more!! 🙌

![IPFS Ecosystem Diagram](https://user-images.githubusercontent.com/618519/74208137-0fc98a00-4c37-11ea-800b-ebd61485d332.png)

> See our [Ecosystem Update from IPFS Camp](https://www.youtube.com/watch?v=jpQnQbfhuBc&feature=youtu.be) for more! 

Author: ```suggestion
```

> This growth caused us to shift some of our attention midyear to support new usage and demand - re-focusing chunks of our team on improving documentation, gateway performance, and our testing tools to validate large-scale network upgrades (see our [Operation Task-Force blog post](https://blog.ipfs.io/2019-07-31-operation-task-force/) for details). We still made progress on our Package Managers goal, but less than we hoped given also investing heavily in other critical areas to support the ecosystem. 🤞

Author: ```suggestion
This growth caused us to shift some of our attention midyear to support new usage and demand - re-focusing some of our working groups on improving documentation, gateway performance, and our testing tools to validate large-scale network upgrades (see our [Operation Task-Force blog post](https://blog.ipfs.io/2019-07-31-operation-task-force/) for details). We still made progress on our Package Managers goal, but less than we hoped given **also** investing heavily in other critical areas to support the ecosystem. 🤞
```

## Our Focus on Package Managers

Our main goal for last year was to improve IPFS performance and scalability by analyzing the needs and pain points for using IPFS in **Package Managers**. This goal was less about _package management_ in particular, and more about defining a representative use case we could research, test, and drive improvements around that would **also** benefit all downstream IPFS users with similar performance and scalability needs for adding, updating, and fetching large datasets. 

> Focusing on a representative use case like package managers brought focus and structure to our prioritization of improvements to IPFS. We built a number of proofs-of-concept (POCs) like [apt-on-ipfs](https://github.com/ipfs-shipyard/apt-on-ipfs), [npm-on-ipfs](https://github.com/ipfs-shipyard/npm-on-ipfs), [clojars-on-ipfs](https://github.com/ipfs-shipyard/clojars-mirror-test) and [homebrew-on-ipfs](https://github.com/ipfs/package-managers/issues/12) to analyze how well IPFS performed against user expectations for adding, updating, and fetching large package repositories—allowing us to identify and fix major pain points. For example, our POCs identified a huge bottleneck in adding GBs of data to IPFS. Further testing resulted in _doubling_ the speed of adding on Linux & OSX devices by switching to async-by-default datastores. While IPFS had previously taken **24mins** to add an Arch Linux repo, after [these fixes](https://github.com/ipfs/go-ipfs/issues/6523#issuecomment-546822191) it only took **11mins** (comparable to the time to copy/paste)! _Note: improvement seen on the badger datastore, already 3x faster than flatfs!_ 

Author: ```suggestion
Focusing on a representative use case like package managers brought focus and structure to our prioritization of improvements to IPFS. We built a number of proofs-of-concept (POCs) like [apt-on-ipfs](https://github.com/ipfs-shipyard/apt-on-ipfs), [npm-on-ipfs](https://github.com/ipfs-shipyard/npm-on-ipfs), [clojars-on-ipfs](https://github.com/ipfs-shipyard/clojars-mirror-test) and [homebrew-on-ipfs](https://github.com/ipfs/package-managers/issues/12) to analyze how well IPFS performed against user expectations for adding, updating, and fetching large package repositories—allowing us to identify and fix major pain points. For example, our POCs identified a huge bottleneck in adding GBs of data to IPFS. Further testing resulted in _doubling_ the speed of adding on Linux & OSX devices by switching to async-by-default datastores.🎉 While IPFS had previously taken **24mins** to add an Arch Linux repo, after [these fixes](https://github.com/ipfs/go-ipfs/issues/6523#issuecomment-546822191) it only took **11mins** (comparable to the time to copy/paste)! _Note: improvement seen on the badger datastore, already 3x faster than flatfs!_ 
```

---
date: 2020-02-07
url: 2020-02-07-our-focus-for-2020
title: Our Focus for 2020
author: Molly Mackinlay
---

_"The mission of IPFS is to create a resilient, upgradeable, open network to preserve and grow humanity’s knowledge."_

It’s a new year, and with it time to update the [IPFS Project Roadmap](https://github.com/ipfs/roadmap) with our focus for 2020. On the IPFS team, we use regular planning cycles to reflect on our progress, zoom out to assess the bigger picture, and realign on the most important work to do next. These cycles help us ensure we stay on track to achieve our mission, and maximize the value and utility we create in the world. 

> > Looking for the TL;DR? Hop right over to the [IPFS Roadmap Repo](https://github.com/ipfs/roadmap) to see our 2020 priority and Epics.

Author: ```suggestion
> Looking for the TL;DR? Hop right over to the [2020 section of our IPFS Roadmap](https://github.com/ipfs/roadmap#2020-priority) to see our focus and epics.
```

## Looking back at our 2019 Roadmap

> Last year, we did a large 2019 Roadmapping effort to write down our mission, define some of the many long-term goals for the IPFS Project, and prioritize where to focus our energy first. We had a lot of “planning” debt, so going from 0 to 1 on a project roadmap was a large endeavor. Our process involved each working group generating a roadmap around a shared goal, and then merging important work streams into “epics” for the whole project that highlighted our major objectives. You can see how we did against these objectives in our [2019 Project Roadmap](https://github.com/ipfs/roadmap), or dive more into how we chose those goals in our [2019 Roadmap blog post](https://blog.ipfs.io/78-ipfs-2019-roadmap).

Author: ```suggestion
Last year, we did a large 2019 Roadmapping effort to write down our mission, define some of the many long-term goals for the IPFS Project, and prioritize where to focus our energy first. We had a lot of “planning” debt, so going from 0 to 1 on a project roadmap was a large endeavor. Our process involved each working group generating a roadmap around a shared goal, and then merging important work streams into “epics” for the whole project that highlighted our major objectives. You can see how we did against these objectives in our [2019 Project Roadmap](https://github.com/ipfs/roadmap/blob/master/2019-IPFS-Project-Roadmap.md), or dive more into how we chose those goals in our [2019 Roadmap blog post](https://blog.ipfs.io/78-ipfs-2019-roadmap).
```

## Our Focus on Package Managers

Our goal for last year was to improve IPFS performance and scalability by analyzing the needs and pain points for using IPFS in **Package Managers**. This goal was less about _package management_ in particular, and more about defining a representative use case that we could research, test, and drive improvements around that would **also** benefit all downstream IPFS users with similar performance and scalability needs for adding, updating, and fetching large datasets. 

> Focusing on a representative use case like package managers brought focus and structure to our prioritization of improvements to IPFS. We built a number of proofs-of-concepts (POCs) like [apt-on-ipfs](https://github.com/ipfs-shipyard/apt-on-ipfs), [npm-on-ipfs](https://github.com/ipfs-shipyard/npm-on-ipfs), [clojars-on-ipfs](https://github.com/ipfs-shipyard/clojars-mirror-test) and [homebrew-on-ipfs](https://github.com/ipfs/package-managers/issues/12) to analyze how well IPFS performed against user expectations for adding, updating, and fetching large package repositories - allowing us to identify and fix major pain points. For example, our POCs identified a huge bottleneck in adding GBs of data to IPFS. Further testing resulted in _doubling_ the speed of adding on Linux & OSX devices by switching to async-by-default datastores. While IPFS had previously taken 24mins to add an Arch Linux repo, after [these fixes](https://github.com/ipfs/go-ipfs/issues/6523#issuecomment-546822191) it only took 11mins (comparable to the time to copy/paste)! _Note: improvement seen on the badger datastore, already 3x faster than flatfs!_ 

Author: @aschmahmann - can you take a look at this and suggest edits?

Our Package Managers goal also helped focus our user research, quarterly objectives, and collaborations. In Q1&2, our Package Managers team surveyed the space and identified core needs documented in our [Package Managers repo](https://github.com/ipfs/package-managers). These insights informed our [quarterly OKR planning](https://github.com/ipfs/team-mgmt/blob/master/OKR/PACKAGE-MANAGERS.md), driving features to make it easier to mirror file system package managers on IPFS. One particular addition was adding metadata on “last updated” time to our unixfs data model to support smarter/faster package updates (humbly named “unixfsv1.5” - already implemented in js-ipfs and coming soon to go). A number of these package-manager-focused improvements are all slated to roll out in our next feature release, go-ipfs 0.5.0 - [feel free to follow along here](https://github.com/ipfs/go-ipfs/issues/6776)!

> We also formed collaborations with IPFS users to partner on improving IPFS for package management use cases. One of our main collaborations was with [Netflix]() to optimize the speed of fetching large container images with Bitswap, our peer-to-peer data transfer algorithm. You can read more about that specific collaboration - and the resulting 3-5x improvements in Bitswap performance! - in [our previous blog post](TODO:NetflixPost). 

Author: TODO

Author: added, but need to be sure to update

IPFS Cluster also released [“collaborative clusters”](https://blog.ipfs.io/2020-01-09-collaborative-clusters/), a new feature to enable package manager maintainers and mirrors to add and replicate repositories across a community of IPFS nodes. With collaborative clusters, any maintainer can push new updates to the pinset of data to mirror, which is then sharded and sync’d across all mirroring nodes. We’ve already seen package managers like Pac-Man added to collaborative clusters along with many “data package managers” like Wikipedia and Project Gutenberg - [follow these instructions to add your own!](https://collab.ipfscluster.io/)

> ## How we did against our goal

Author: ```suggestion
## How We Did Against our Goal
```

We made huge strides this year on the performance at scale needed by many package manager communities, but it’s clear there’s still lots of work left to be done to achieve widespread adoption and meet the needs of more types of package managers. While focusing on a specific use case did help us identify and drive important fixes, it didn’t give us feedback across the entire IPFS ecosystem as to what pain points were top priority blockers to growth or value-add. We also found that a number of paths to increasing IPFS usage in package managers were actually divergent from the core goal we wanted to achieve: making IPFS itself better. This tension required constant vigilance to ensure features and improvements would be felt by many downstream use cases - not just specialized package manager tooling. 

> In the end, we succeeded at staying focused, not building another new package manager, and landing a ton of features and improvements to make IPFS better for everyone — where we didn’t succeed was shipping and integrating those improvements into existing package managers to drive adoption and visibility of IPFS with an awesome community of aligned and talented developers. With the features and improvements landing in 0.5.0, that future adoption work is significantly unblocked - however we also learned through our research that most package manager communities aren’t fast adopters of new tooling around package distribution due to the semi-pro-bono nature of maintainership. To support a slower, more ad-how adoption cadence, we plan to use channels like DevGrants and Collaborations to support package manager adopters acting within their own communities. This allows us to continue focusing our core working groups on improving and augmenting the core protocol and reference implementation, while supporting many community applications and refinements of these tools. Excited to use these new improvements to IPFS in your package management use case? [Make an issue in the package managers repo](https://github.com/ipfs/package-managers/issues) for the latest recommendations!

Author: @andrew does this seem right to you as the previous steward of our package managers work? Am I communicating this accurately to our package manager communities?

Reviewer: Yeah this is great, nicely sums it up and still leaves the door wide open for future package manager integrations and collabs 👍

Author: Thanks!

- [Content Routing ZenHub Roadmap](https://app.zenhub.com/workspaces/content-routing-2020-5e29be25ab9d8da0b032cfcc/roadmap)
- [Testground ZenHub Roadmap](https://app.zenhub.com/workspaces/testground-5db6a5bf7ca61c00014e2961/roadmap)

We considered a number of other potential goals - especially all the great [2020 Theme Proposals](https://github.com/ipfs/roadmap/issues/) - before selecting this priority. However, we decided it was more important to focus core working group dev time on the main blockers and pain points to enable the entire ecosystem to grow and succeed. Many of these proposals are actually very well suited for community ownership via [DevGrants](https://github.com/ipfs/devgrants) and collaborations - and some of them, like ["IPFS in Rust"](https://github.com/ipfs/roadmap/issues/54) and ["Examples and Tutorials"](https://github.com/ipfs/roadmap/issues/50), already have grants or bounties associated with them!

> ## Increasing contributor velocity while supporting adoption

Author: ```suggestion
## Increasing Contributor Velocity While Supporting Adoption
```

Part of allowing our core go-ipfs developers to focus deeply on content routing is to systematize and improve our maintainership for the many IPFS users and contributors merging changes into the core repos. As we’ve grown as a project, we’ve not done the necessary work of intentionally distributing and decentralizing this stewardship, in order to make community contribution scalable too. We want to create more pathways for folks throughout the community to step up to help with this, and build better avenues for experimentation with fast feedback loops.

> We also want to grow and support the many amazing contributors to ipfs, and exciting new explorations to build tools or address new use cases with IPFS. There were a ton of amazing suggestions for features and focus areas in our 2020 theme collection that would be immensely valuable for the project, but our core working groups can’t push forward this quarter. To help with that, we’re launching a community DevGrants program where Protocol Labs and other groups can offer bounties, RFPs, and accept open suggestions for improvements, new features, and even new implementations. To kick this off, we’ve added XX bounties suggested by our 2020 Theme proposal process, and merged X RFPs or open grant proposals for larger-scale improvements to help benefit the whole community. If you are a user or contributor to IPFS, this is a great opportunity to both sponsor and apply for [grants](https://github.com/ipfs/devgrants/tree/master/rfps#how-to-create-rfp) or [bounties](https://github.com/ipfs/devgrants/blob/master/BOUNTIES.md#how-to-propose) to help make IPFS better for everyone! 

Author: TODO: update with fresh #s

To help coordinate this work, we’re creating a new “Ecosystem” working group made up of 3 Special Interest Groups (SIGs) focused on developer experience, collabs and community, and browsers/connectivity. Our Ecosystem working group’s objective is to ensure community health and growth through collaborations, developer experience and platform availability. The 3 SIGs each focus on:
**Browsers & Connectivity**: Maximize availability & connectivity of IPFS on the web
**Collabs & Community**: Support IPFS users and grow new opportunities through research, collaborations and community engagement
**Developer Experience & Maintainership**: Support the IPFS technical community through documentation, contributor experience, API ergonomics and tooling

Curious about the work they’re taking on? Check out the [Project Roadmap Epics](https://github.com/ipfs/roadmap#2020-epics) for more tangible goals, and be on the lookout for more suggestions on how you can help enable other contributors throughout the IPFS ecosystem!

> ## Everything else

Author: ```suggestion
## Everything Else
```

---

> date: 2020-02-07

Author: update =[

Focusing on a representative use case like package managers brought focus and structure to our prioritization of improvements to IPFS. We built a number of proofs-of-concept (POCs) like [apt-on-ipfs](https://github.com/ipfs-shipyard/apt-on-ipfs), [npm-on-ipfs](https://github.com/ipfs-shipyard/npm-on-ipfs), [clojars-on-ipfs](https://github.com/ipfs-shipyard/clojars-mirror-test) and [homebrew-on-ipfs](https://github.com/ipfs/package-managers/issues/12) to analyze how well IPFS performed against user expectations for adding, updating, and fetching large package repositories—allowing us to identify and fix major pain points. For example, our POCs identified a huge bottleneck in adding GBs of data to IPFS. Further testing resulted in _doubling_ the speed of adding on Linux & OSX devices by switching to async-by-default datastores. While IPFS had previously taken 24mins to add an Arch Linux repo, after [these fixes](https://github.com/ipfs/go-ipfs/issues/6523#issuecomment-546822191) it only took 11mins (comparable to the time to copy/paste)! _Note: improvement seen on the badger datastore, already 3x faster than flatfs!_ 

> Our Package Managers goal also helped focus our user research, quarterly objectives, and collaborations. In Q1&2, our Package Managers team surveyed the space and identified core needs documented in our [Package Managers repo](https://github.com/ipfs/package-managers). These insights informed our [quarterly OKR planning](https://github.com/ipfs/team-mgmt/blob/master/OKR/PACKAGE-MANAGERS.md), driving features to make it easier to mirror file system package managers on IPFS. One particular addition was adding metadata on “last updated” time to our unixfs data model to support smarter/faster package updates (humbly named “unixfsv1.5” and already implemented in js-ipfs; coming soon to go). A number of these package-manager-focused improvements are all slated to roll out in our next feature release, go-ipfs 0.5.0 - [feel free to follow along here](https://github.com/ipfs/go-ipfs/issues/6776)!

Reviewer 2: We might want to make it clear that none of these changes have been released (e.g., you'll need the latest master to get the add speed improvements).

Author: added a line for grabbing master to try them out

We also formed collaborations with IPFS users to partner on improving IPFS for package management use cases. One of our main collaborations was with [Netflix](https://netflix.com) to optimize the speed of fetching large container images with [Bitswap](https://github.com/ipfs/go-bitswap), our peer-to-peer data transfer algorithm. You can read more about that specific collaboration—and the resulting 3-5x improvements in Bitswap performance!—in [our previous blog post](/2020-02-10-improved-bitswap-with-netflix/). 

> 

Author: reminder to me to check this link before shipping >.<

## How We Did Against Our Goal

We made huge strides this year on the performance at scale needed by many package manager communities, but it’s clear there’s still lots of work left to be done to achieve widespread adoption and meet the needs of more types of package managers. While focusing on a specific use case did help us identify and drive important fixes, it didn’t give us feedback across the entire IPFS ecosystem as to what pain points were top priority blockers to growth or value-add. We also found that a number of paths to increasing IPFS usage in package managers were actually divergent from the core goal we wanted to achieve: making IPFS itself better. This tension required constant vigilance to ensure features and improvements would be felt by many downstream use cases—not just specialized package manager tooling. 

> In the end, we succeeded at staying focused, not building another new package manager, and landing a ton of features and improvements to make IPFS better for everyone; where we didn’t succeed was shipping and integrating those improvements into existing package managers to drive adoption and visibility of IPFS with an awesome community of aligned and talented developers. With the features and improvements landing in 0.5.0, that future adoption work is significantly unblocked, however we also learned through our research that most package manager communities aren’t fast adopters of new tooling around package distribution due to the semi-pro-bono nature of maintainership. To support a slower, more ad-hoc adoption cadence, we plan to use channels like [DevGrants](https://github.com/ipfs/devgrants) and Collaborations to support package manager adopters acting within their own communities. This allows us to continue focusing our core working groups on improving and augmenting the core protocol and reference implementation, while supporting many community applications and refinements of these tools. 

Reviewer 3: Just to clarify, this is go-ipfs 0.5.0 right?

Author: yep! Should I clarify/link again?

Author: ```suggestion
In the end, we succeeded at staying focused, not building another new package manager, and landing a ton of features and improvements to make IPFS better for everyone; where we didn’t succeed was shipping and integrating those improvements into existing package managers to drive adoption and visibility of IPFS with an awesome community of aligned and talented developers. With the [features and improvements landing in 0.5.0](https://github.com/ipfs/go-ipfs/issues/6776), that future adoption work is significantly unblocked, however we also learned through our research that most package manager communities aren’t fast adopters of new tooling around package distribution due to the semi-pro-bono nature of maintainership. To support a slower, more ad-hoc adoption cadence, we plan to use channels like [DevGrants](https://github.com/ipfs/devgrants) and Collaborations to support package manager adopters acting within their own communities. This allows us to continue focusing our core working groups on improving and augmenting the core protocol and reference implementation, while supporting many community applications and refinements of these tools. 
```

Reviewer 3: Much better!

## 2020 Roadmapping Process

We did a slightly different roadmapping process this year, after learning from the wins and challenges we encountered with our massive roadmapping endeavor in 2019. Rather than starting with project goals to inspire a distributed working group roadmapping process (which was difficult to merge and hard to make open), we decided to flip the order of things and start with a [broad call for themes and suggestions](https://github.com/ipfs/roadmap/blob/master/2020-IPFS-Project-Planning.md) from the community for where the IPFS project should focus in 2020. 

We got [**11** awesome proposals](https://github.com/ipfs/roadmap/issues), with great ideas and discussion across a wide variety of tracks. Notable mentions include [IPFS in Rust](https://github.com/ipfs/roadmap/issues/54), [IPFS for Mobile](https://github.com/ipfs/roadmap/issues/45), [IPFS-Wikipedia](https://github.com/ipfs/roadmap/issues/46), and [IPFS-powered GitHub](https://github.com/ipfs/roadmap/issues/43). Huge thank you for the community members who took the time to write these proposals, and call out the core needs and gaps, which fed directly into our next stage!

> <blockquote class="twitter-tweet"><p lang="en" dir="ltr">Huge thanks to <a href="https://twitter.com/agentofuser?ref_src=twsrc%5Etfw">@agentofuser</a> <a href="https://twitter.com/dietrich?ref_src=twsrc%5Etfw">@dietrich</a> <a href="https://twitter.com/Ianopolous?ref_src=twsrc%5Etfw">@Ianopolous</a> <a href="https://twitter.com/lidelOrg?ref_src=twsrc%5Etfw">@lidelOrg</a> <a href="https://twitter.com/NukeManDan?ref_src=twsrc%5Etfw">@NukeManDan</a> <a href="https://twitter.com/aschmahmann?ref_src=twsrc%5Etfw">@aschmahmann</a> <a href="https://twitter.com/vasa_develop?ref_src=twsrc%5Etfw">@vasa_develop</a> &amp; <a href="https://twitter.com/hashtag/liyunzhi_666?src=hash&amp;ref_src=twsrc%5Etfw">#liyunzhi_666</a> for the amazing contributions. There’s far more here than one team can build in 2020, but all awesome ideas &amp; much we can empower the whole community to pitch in on! <a href="https://t.co/jUmEJ6nd3k">pic.twitter.com/jUmEJ6nd3k</a></p>&mdash; Molly (@momack28) <a href="https://twitter.com/momack28/status/1193404771983454214?ref_src=twsrc%5Etfw">November 10, 2019</a></blockquote> <script async src="https://platform.twitter.com/widgets.js" charset="utf-8"></script>

Reviewer 4: 11 proposals noted above, but only 8 ppl listed here - if listing, should list all (including you!)

Author: rust and offline-first came after this tweet (https://twitter.com/momack28/status/1193404771983454214) - I can fake the rewrite here, but I can't change the actual tweet...

Author: ```suggestion
<blockquote class="twitter-tweet"><p lang="en" dir="ltr">Huge thanks to <a href="https://twitter.com/agentofuser?ref_src=twsrc%5Etfw">@agentofuser</a> <a href="https://twitter.com/dietrich?ref_src=twsrc%5Etfw">@dietrich</a> <a href="https://twitter.com/Ianopolous?ref_src=twsrc%5Etfw">@Ianopolous</a> <a href="https://twitter.com/lidelOrg?ref_src=twsrc%5Etfw">@lidelOrg</a> <a href="https://twitter.com/NukeManDan?ref_src=twsrc%5Etfw">@NukeManDan</a> <a href="https://twitter.com/aschmahmann?ref_src=twsrc%5Etfw">@aschmahmann</a> <a href="https://twitter.com/vasa_develop?ref_src=twsrc%5Etfw">@vasa_develop</a> <a href="https://twitter.com/hashtag/liyunzhi_666?src=hash&amp;ref_src=twsrc%5Etfw">#liyunzhi_666</a> <a href="https://twitter.com/hashtag/gozala?src=hash&amp;ref_src=twsrc%5Etfw">@gozala</a> &amp; <a href="https://twitter.com/hashtag/parkan?src=hash&amp;ref_src=twsrc%5Etfw">@parkan</a> for the amazing contributions. There’s far more here than one team can build in 2020, but all awesome ideas &amp; much we can empower the whole community to pitch in on! <a href="https://t.co/jUmEJ6nd3k">pic.twitter.com/jUmEJ6nd3k</a></p>&mdash; Molly (@momack28) <a href="https://twitter.com/momack28/status/1193404771983454214?ref_src=twsrc%5Etfw">November 10, 2019</a></blockquote> <script async src="https://platform.twitter.com/widgets.js" charset="utf-8"></script>
```

## Increasing Contributor Velocity While Supporting Adoption

Part of allowing our core go-ipfs developers to focus deeply on content routing is to systematize and improve our maintainership for the many IPFS users and contributors merging changes into the core repos. As we’ve grown as a project, we’ve not done the necessary work of intentionally distributing and decentralizing this stewardship, in order to make community contribution scalable too. We want to create more pathways for folks throughout the community to step up to help with this, and build better avenues for experimentation with fast feedback loops.

We also want to grow and support the many amazing contributors to IPFS, and exciting new explorations to build tools or address new use cases. There were a ton of amazing suggestions for features and focus areas in our 2020 theme collection that would be immensely valuable for the project, but our core working groups can’t push forward this quarter. To help with that, we’re launching a community DevGrants program where Protocol Labs and other groups can offer bounties, RFPs, and accept open suggestions for improvements, new features, and even new implementations. 

> To kick this off, we’ve added XX bounties suggested by our 2020 Theme proposal process, and merged X RFPs or open grant proposals for larger-scale improvements to help benefit the whole community. If you are a user or contributor to IPFS, this is a great opportunity to both sponsor and apply for [grants](https://github.com/ipfs/devgrants/tree/master/rfps#how-to-create-rfp) or [bounties](https://github.com/ipfs/devgrants/blob/master/BOUNTIES.md#how-to-propose) to help make IPFS better for everyone! 

Reviewer 2: XX -> TODO (so it's not lost)

Reviewer 2: (oh, already marked)
